src/s3path.jl

```diff
@@ -4,17 +4,18 @@ struct S3Path{A<:AbstractAWSConfig} <: AbstractPath
     drive::String
     isdirectory::Bool
     config::A
+    version::String
```

Comment:
Verified strings are appropriate. https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html#version-ids

Possibly we may want to use `nothing` for unversioned/versioned unaware `S3Path`s

Reply:
After reading over the entire thing I think using `nothing` is a better option
```suggestion
    version::Union{String,Nothing}
```
You may want to have:
```julia
const S3PathVersion = Union{String,Nothing}
```

Reply:
Updated!